src/Components/Aspire.Keycloak/README.md

```diff
@@ -0,0 +1,105 @@
+# Aspire.Keycloak library
+
+Adds JwtBearer and OpenId Connect authentication to the project via a [Keycloak](https://www.keycloak.org).
+
+## Getting started
+
+### Prerequisites
+
+- A Keycloak server instance
+- A Keycloak realm
+- For JwtBearer authentication, a configured audience in the Keycloak realm
+- For OpenId Connect authentication, the ID of a client configured in the Keycloak realm
+
+### Install the package
+
+Install the .NET Aspire Keycloak library with [NuGet](https://www.nuget.org):
+
+```dotnetcli
+dotnet add package Aspire.Keycloak
+```
+
+## Jwt bearer authentication usage example
+
+In the _Program.cs_ file of your ASP.NET Core API project, call the `AddKeycloakJwtBearer` extension method to add JwtBearer authentication, using a connection name, realm and any required JWT Bearer options:
+
+```csharp
+builder.Services.AddAuthentication()
+                .AddKeycloakJwtBearer("keycloak", realm: "WeatherShop", options =>
+                {
+                    options.Audience = "weather.api";
+                });
+```
+
+You can set many other options via the `Action<JwtBearerOptions> configureOptions` delegate.
+
+## OpenId Connect authentication usage example
+
+In the _Program.cs_ file of your Blazor project, call the `AddKeycloakOpenIdConnect` extension method to add OpenId Connect authentication, using a connection name, realm and any required OpenId Connect options:
+
+```csharp
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+                .AddKeycloakOpenIdConnect(
+                    "keycloak", 
+                    realm: "WeatherShop", 
+                    options =>
+                    {
+                        options.ClientId = "WeatherWeb";
+                        options.ResponseType = OpenIdConnectResponseType.Code;
+                        options.Scope.Add("weather:all");
+                    });
+```
+
+You can set many other options via the `Action<OpenIdConnectOptions>? configureOptions` delegate.
+
+## AppHost extensions
+
+In your AppHost project, install the `Aspire.Hosting.Keycloak` library with [NuGet](https://www.nuget.org):
+
+```dotnetcli
+dotnet add package Aspire.Hosting.Keycloak
+```
+
+Then, in the _Program.cs_ file of `AppHost`, register a Keycloak server and consume the connection using the following methods:
+
+```csharp
+var keycloak = builder.AddKeycloak("keycloak");
+
+var apiService = builder.AddProject<Projects.Keycloak_ApiService>("apiservice")
+                        .WithReference(keycloak);
+
+builder.AddProject<Projects.Keycloak_Web>("webfrontend")
+    .WithExternalHttpEndpoints()
+    .WithReference(keycloak)
+    .WithReference(apiService);
```

Comment:
```suggestion
builder.AddProject<Projects.Keycloak_Web>("webfrontend")
        .WithExternalHttpEndpoints()
        .WithReference(keycloak)
        .WithReference(apiService);
```

To be consistent with the spacing elsewhere in this README.

Reply:
"thoughts on the naming here?"

@eerhardt The naming of the project? Sounds OK to me?

"I think we should have it in both READMEs, so people can connect the 2 packages (one for the AppHost and one for their projects). Or at least point to which NuGet package should be used in the apps here."

This is how we do the README of all other hosting resources. You mean copy the README from the client component here?

Reply:
> The naming of the project? Sounds OK to me?

Yes, the name of the project is the name of the NuGet package. Aspire.Keycloak is a pretty general name. @DamianEdwards suggested `Aspire.Keycloak.Authentication`.

> This is how we do the README of all other hosting resources. You mean copy the README from the client component here?

No, copying the whole client component README doesn't make sense. I guess if this is the way the other hosting READMEs are (with no link to the corresponding Aspire client library), then we can keep it as it is.

Reply:
@eerhardt Renamed the project to `Aspire.Keycloak.Authentication`